pkg/api-server/config_ws_test.go

```diff
@@ -0,0 +1,55 @@
+package api_server_test
+
+import (
+	"fmt"
+	"github.com/Kong/kuma/pkg/config"
+	api_server_config "github.com/Kong/kuma/pkg/config/api-server"
+	kuma_cp "github.com/Kong/kuma/pkg/config/app/kuma-cp"
+	"github.com/Kong/kuma/pkg/plugins/resources/memory"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"io/ioutil"
+	"net/http"
+)
+
+var _ = Describe("Config WS", func() {
+
+	It("should return the config", func() {
+		// given
+		cfg := api_server_config.DefaultApiServerConfig()
+
+		// setup
+		resourceStore := memory.NewStore()
+		apiServer := createTestApiServer(resourceStore, cfg)
+
+		stop := make(chan struct{})
+		go func() {
+			defer GinkgoRecover()
+			err := apiServer.Start(stop)
+			Expect(err).ToNot(HaveOccurred())
+		}()
+
+		// wait for the server
+		Eventually(func() error {
+			_, err := http.Get(fmt.Sprintf("http://localhost%s/config", apiServer.Address()))
+			return err
+		}, "3s").ShouldNot(HaveOccurred())
+
+		// when
+		resp, err := http.Get(fmt.Sprintf("http://localhost%s/config", apiServer.Address()))
+		Expect(err).ToNot(HaveOccurred())
+
+		// then
+		body, err := ioutil.ReadAll(resp.Body)
+		Expect(err).ToNot(HaveOccurred())
+
+		// when
+		expectedConfig := kuma_cp.DefaultConfig()
+		expectedConfig.ApiServer = cfg
+		cfgJson, err := config.ConfigForDisplayJson(&expectedConfig)
+		Expect(err).ToNot(HaveOccurred())
+
+		// then
+		Expect(cfgJson).To(MatchJSON(body))
```

Comment:
It's strange that actual server response is treated as "expected" value. And the "actual" value is computed by using the same function that is being tested (`config.ConfigForDisplayJson`).

It's completely unclear that the goal of this test is to verify that all secrets have been removed from `/config` response.

Let's make it more apparent by hardcoding the exact "expected" value.

Reply:
I wanted to avoid creating the test that will have to be modified with every config change.

If we are fine with this, I can hardcode expected value.

Reply:
Yeah, it make sense to hardcode this value. This way we test the json output and sanitizing values.